tests/test_hash_table.py

```diff
@@ -0,0 +1,75 @@
+import unittest
+
+from src.hash_table import ChainHashTable
+
+
+class TestChainHashTable(unittest.TestCase):
+    def test_reference_init(self):
+        ht = ChainHashTable(8)
+        ht._ChainHashTable__bucket[0].val = 1
+        self.assertFalse(ht._ChainHashTable__bucket[1].val == 1)
+
+    def test_search(self):
+        ht = ChainHashTable(8)
+        ht.insert("3", "a")
+        self.assertEqual(ht.search("3"), "a")
+        ht.insert("3", "b")
+        self.assertEqual(ht.search("3"), "b")
+        ht.insert("11", "c")
+        self.assertEqual(ht.search("11"), "c")
+        # Key not exist
+        self.assertEqual(ht.search("88"), None)
+
+    def test_remove(self):
+        ht = ChainHashTable(8)
+        # Remove from empty hash table should return None
+        self.assertEqual(ht.remove("8"), None)
+        # Remove from empty should leave size == 0 rather than -1
+        self.assertEqual(ht.size(), 0)
```

Comment:
IMO, this method needs to test for normal case. For example, when we remove keys which exist.

Reply:
I included tests for the normal cases in `test_all` method. I intended to only create new test method for edge cases, so I will change the name `test_all` to `test_normal`.